cfme/tests/infrastructure/test_infra_quota_via_services.py

```diff
@@ -0,0 +1,274 @@
+        ele_name=fauxfactory.gen_alphanumeric(),
+        ele_desc="my ele desc",
+        choose_type="Text Box",
+        default_text_box="default value"
+    )
+    service_dialog = ServiceDialog(label=dialog, description="my dialog",
+                                   submit=True, cancel=True,
+                                   tab_label="tab_" + fauxfactory.gen_alphanumeric(),
+                                   tab_desc="my tab desc",
+                                   box_label="box_" + fauxfactory.gen_alphanumeric(),
+                                   box_desc="my box desc")
+    service_dialog.create(element_data)
+    flash.assert_success_message('Dialog "{}" was added'.format(dialog))
+    yield dialog
+
+
+@pytest.yield_fixture(scope="function")
+def catalog():
+    catalog = "cat_" + fauxfactory.gen_alphanumeric()
+    cat = Catalog(name=catalog,
+                  description="my catalog")
+    cat.create()
+    yield catalog
+
+
+@pytest.yield_fixture(scope="module")
+def set_tenant_cpu():
+    cpu_data = {'cpu_cb': True, 'cpu': 2}
+    reset_cpu_data = {'cpu_cb': False}
+    roottenant = Tenant.get_root_tenant()
+    roottenant.set_quota(**cpu_data)
+    yield
+    roottenant.set_quota(**reset_cpu_data)
+
+
+@pytest.yield_fixture(scope="module")
+def set_tenant_memory():
+    memory_data = {'memory_cb': True, 'memory': 2}
+    reset_memory_data = {'memory_cb': False}
+    roottenant = Tenant.get_root_tenant()
+    roottenant.set_quota(**memory_data)
+    yield
+    roottenant.set_quota(**reset_memory_data)
+
+
+@pytest.yield_fixture(scope="module")
+def set_tenant_storage():
+    storage_data = {'storage_cb': True, 'storage': 1}
+    reset_storage_data = {'storage_cb': False}
+    roottenant = Tenant.get_root_tenant()
+    roottenant.set_quota(**storage_data)
+    yield
+    roottenant.set_quota(**reset_storage_data)
+
+
+@pytest.yield_fixture(scope="module")
+def set_tenant_vm():
+    vm_data = {'vm_cb': True, 'vm': 1}
+    reset_vm_data = {'vm_cb': False}
+    roottenant = Tenant.get_root_tenant()
+    roottenant.set_quota(**vm_data)
+    yield
+    roottenant.set_quota(**reset_vm_data)
+
+
+@pytest.fixture(scope="function")
+def provisioning_data(provider, vm_name, provisioning):
+    template, host, datastore, iso_file, catalog_item_type, vlan = map(provisioning.get,
+        ('template', 'host', 'datastore', 'iso_file', 'catalog_item_type', 'vlan'))
+
+    provisioning_data = {
+        'vm_name': vm_name,
+        'host_name': {'name': [host]},
+        'datastore_name': {'name': [datastore]},
+        'vlan': vlan,
+        'template': template,
+        'provision_type': "Native Clone" if provider.type == "rhevm" else "VMware",
+        'catalog_item_type': catalog_item_type,
+        'item_name': fauxfactory.gen_alphanumeric()
+    }
+    return provisioning_data
+
+
+@pytest.mark.tier(1)
+def test_tenant_infra_cpu_quota_via_services(provisioning_data, provider, setup_provider, request,
```

Comment:
**Optional**
Since these tests have the same (or close to) fixture use, consider using pytest.mark.usefixtures() at the beginning of the test to shorten these method definitions.